fdbserver/ClusterController.actor.cpp

```diff
@@ -2197,10 +2197,10 @@ class ClusterControllerData {
 		    db.recoveryStalled) {
 			if (db.config.regions.size() > 1) {
 				auto regions = db.config.regions;
-				if (clusterControllerDcId.get() == regions[0].dcId) {
+				if (clusterControllerDcId.get() == regions[0].dcId && regions[1].priority >= 0) {
 					std::swap(regions[0], regions[1]);
 				}
-				ASSERT(clusterControllerDcId.get() == regions[1].dcId);
+				ASSERT(regions[1].priority < 0 || clusterControllerDcId.get() == regions[1].dcId);
```

Comment:
After failover, why the original primary's priority is smaller than 0?

Reply:
This is to account for the case where a failover is not triggered. This assertion checks that either (1) remote is disabled so that there is no failover, or (2) remote is enabled, and failover is definitely triggered (by checking that the cluster controller is now running at the remote DC.